src/theory/logic_info.h

```diff
@@ -123,6 +123,9 @@ class CVC5_EXPORT LogicInfo
   /** Is this the all-inclusive logic? */
   bool hasEverything() const;
 
+  /** Is this the all-inclusive higher-order logic? */
```

Comment:
I don't like having this method.  Can we just make `hasEverything` robust to whether HO is enabled? Then, this method is equivalent to `hasEverything() && isHigherOrder()`?

Reply:
Yes, I recommend changing how hasEverything is implemented.  It would check "ALL" or "HO_ALL" I suppose.

Reply:
OK, I've done that in the last commit.